### PR TITLE
[PhpDocParser] Make use of native attributes

### PR DESCRIPTION
**Pull request by TomasVotruba**
Waits on https://github.com/phpstan/phpdoc-parser/pull/57 to be merged.

Then it's ready to :rocket:

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -87,7 +87,25 @@
...
+            "vendor/phpstan/phpdoc-parser/src/Ast/PhpDoc/PhpDocTextNode.php",
+            "vendor/phpstan/phpdoc-parser/src/Ast/PhpDoc/ParamTagValueNode.php",
+            "vendor/phpstan/phpdoc-parser/src/Ast/PhpDoc/VarTagValueNode.php",
+            "vendor/phpstan/phpdoc-parser/src/Ast/Type/ArrayTypeNode.php",
+            "vendor/phpstan/phpdoc-parser/src/Ast/Type/ArrayShapeNode.php",
+            "vendor/phpstan/phpdoc-parser/src/Ast/Type/ArrayShapeItemNode.php",
+            "vendor/phpstan/phpdoc-parser/src/Ast/Type/GenericTypeNode.php",
+            "vendor/phpstan/phpdoc-parser/src/Ast/Type/CallableTypeNode.php",
+            "vendor/phpstan/phpdoc-parser/src/Ast/Type/CallableTypeParameterNode.php"
+        ]
```
Comment (TomasVotruba) [inline composer.json:108]: This is require to give the fork vendor/phpdoc-parser priority over internal `phpstan.phar` ones. Resp. to give priority `BaseNode` with attributes over none.

**Comment by TomasVotruba** [comment]
Now we have to wait till PHPStan is released with new php-doc-parser version, because it takes priority in autoload now.


image

**Comment by ondrejmirtes** [comment]
FYI I just tagged PHPStan 0.12.82 that contains phpdoc-parser 0.5. (Wait a few moments till it shows up as a release.)

**Comment by TomasVotruba** [comment]
There is a BC break in https://github.com/phpstan/phpdoc-parser/commit/b451b2f27a97c8c288de63db79c2a843727d4326

Trying to figure out the fix now

**Comment by ondrejmirtes** [comment]
What BC break?

File: composer.json [hunk 2]
```diff
@@ -52,32 +52,32 @@
...
-        "symplify/set-config-resolver": "^9.2",
-        "symplify/simple-php-doc-parser": "^9.2",
-        "symplify/skipper": "^9.2",
-        "symplify/smart-file-system": "^9.2",
-        "symplify/symfony-php-config": "^9.2",
+        "symplify/astral": "^9.2.12",
+        "symplify/autowire-array-parameter": "^9.2.12",
+        "symplify/console-color-diff": "^9.2.12",
+        "symplify/package-builder": "^9.2.12",
+        "symplify/rule-doc-generator-contracts": "^9.2.12",
```
Comment (samsonasik) [inline composer.json:59]: seems this will fix https://github.com/rectorphp/rector/issues/5908 👍

**Comment by TomasVotruba** [comment]
@ondrejmirtes Format preserving with spaces. We're trying to detect old spaces from old code with regular expression and it now fails, because it's one less.

image

**Pull request merged**